test/Microsoft.AspNetCore.Mvc.IntegrationTests/TryValidateModelIntegrationTest.cs

```diff
@@ -114,28 +117,37 @@ public void TryValidateModel_CollectionsModel_ReturnsErrorsForInvalidProperties(
             Assert.False(result);
             Assert.False(modelState.IsValid);
             var modelStateErrors = GetModelStateErrors(modelState);
+
+            // We define the "CompanyName null" message locally, so we should manually check its value.
+            var categoryRequired = ValidationAttributeUtil.GetRequiredErrorMessage("Category");
+            var priceRange = ValidationAttributeUtil.GetRangeErrorMessage(20, 100, "Price");
+            var contactUsMax = ValidationAttributeUtil.GetStringLengthErrorMessage(null, 20, "Contact Us");
+            var contactusReqex = ValidationAttributeUtil.GetRegExErrorMessage("^[0-9]*$", "Contact Us");
+
             Assert.Equal("CompanyName cannot be null or empty.", modelStateErrors["[0].CompanyName"]);
-            Assert.Equal("The field Price must be between 20 and 100.", modelStateErrors["[0].Price"]);
-            Assert.Equal(
-                PlatformNormalizer.NormalizeContent("The Category field is required."),
-                modelStateErrors["[0].Category"]);
+            Assert.Equal(priceRange, modelStateErrors["[0].Price"]);
+            Assert.Equal(categoryRequired, modelStateErrors["[0].Category"]);
             AssertErrorEquals(
-                "The field Contact Us must be a string with a maximum length of 20." +
-                "The field Contact Us must match the regular expression " +
-                (TestPlatformHelper.IsMono ? "^[0-9]*$." : "'^[0-9]*$'."),
+                contactUsMax +
+                    contactusReqex,
                 modelStateErrors["[0].Contact"]);
             Assert.Equal("CompanyName cannot be null or empty.", modelStateErrors["[1].CompanyName"]);
-            Assert.Equal("The field Price must be between 20 and 100.", modelStateErrors["[1].Price"]);
-            Assert.Equal(
-                PlatformNormalizer.NormalizeContent("The Category field is required."),
-                modelStateErrors["[1].Category"]);
+            Assert.Equal(priceRange, modelStateErrors["[1].Price"]);
+            Assert.Equal(categoryRequired, modelStateErrors["[1].Category"]);
             AssertErrorEquals(
-                "The field Contact Us must be a string with a maximum length of 20." +
-                "The field Contact Us must match the regular expression " +
-                (TestPlatformHelper.IsMono ? "^[0-9]*$." : "'^[0-9]*$'."),
+                contactUsMax +
+                    contactusReqex,
                 modelStateErrors["[1].Contact"]);
         }
 
+        private void AssertErrorEquals(string expected, string actual)
```

Comment:
The comment claims that the order may vary depending on platform / client. I assume they knew what they were talking about so I just left it since it's not hurting anything.